Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SpatialCoordinateSystemManager.cs

```diff
@@ -53,188 +36,265 @@ public class SpatialCoordinateSystemManager : Singleton<SpatialCoordinateSystemM
+            if (participants.TryGetValue(endpoint, out var participant))
             {
-                if (StateSynchronizationObserver.IsInitialized)
-                {
-                    StateSynchronizationObserver.Instance.Connected += OnConnected;
-                    StateSynchronizationObserver.Instance.Disconnected += OnDisconnected;
-                    StateSynchronizationObserver.Instance.RegisterCommandHandler(command, HandleCommand);
-                }
-                if (StateSynchronizationBroadcaster.IsInitialized)
-                {
-                    StateSynchronizationBroadcaster.Instance.Connected += OnConnected;
-                    StateSynchronizationBroadcaster.Instance.Disconnected += OnDisconnected;
-                    StateSynchronizationBroadcaster.Instance.RegisterCommandHandler(command, HandleCommand);
-                }
+                Destroy(participant.gameObject);
+                participants.Remove(endpoint);
             }
         }
 
-        private void UnregisterCommands()
+        private void OnCoordinateStateReceived(SocketEndpoint socketEndpoint, string command, BinaryReader reader, int remainingDataSize)
         {
-            DebugLog($"Unregistering for appropriate commands: StateSynchronizationObserver.IsInitialized: {StateSynchronizationObserver.IsInitialized}, StateSynchronizationBroadcaster.IsInitialized: {StateSynchronizationBroadcaster.IsInitialized}");
-            foreach (var command in supportedCommands)
+            if (!participants.TryGetValue(socketEndpoint, out SpatialCoordinateSystemParticipant participant))
             {
-                if (StateSynchronizationObserver.IsInitialized)
-                {
-                    StateSynchronizationObserver.Instance.Connected -= OnConnected;
-                    StateSynchronizationObserver.Instance.Disconnected -= OnDisconnected;
-                    StateSynchronizationObserver.Instance.UnregisterCommandHandler(command, HandleCommand);
-                }
-                if (StateSynchronizationBroadcaster.IsInitialized)
-                {
-                    StateSynchronizationBroadcaster.Instance.Connected -= OnConnected;
-                    StateSynchronizationBroadcaster.Instance.Disconnected -= OnDisconnected;
-                    StateSynchronizationBroadcaster.Instance.UnregisterCommandHandler(command, HandleCommand);
-                }
+                Debug.LogError($"Failed to find a SpatialCoordinateSystemParticipant for an attached SocketEndpoint");
+                return;
             }
+
+            participant.PeerDeviceHasTracking = reader.ReadBoolean();
+            participant.PeerSpatialCoordinateIsLocated = reader.ReadBoolean();
+            participant.PeerIsLocatingSpatialCoordinate = reader.ReadBoolean();
+            participant.PeerSpatialCoordinateWorldPosition = reader.ReadVector3();
+            participant.PeerSpatialCoordinateWorldRotation = reader.ReadQuaternion();
```

Comment:
i think this logic should be kept in the SpatialCoordinateSystemParticipant for consistency.